app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb

```diff
@@ -213,16 +213,21 @@ def infer_related_host_ems_refs_db!
     changed_hosts.each do |host|
       add_simple_target!(:ems_clusters, uuid_from_target(host.ems_cluster))
       host.storages.each do |storage|
-        add_simple_target!(:storagedomains, uuid_from_target(storage))
+        add_simple_target!(:storagedomains, storage.ems_ref)
+      end
+      host.switches.each do |switch|
+        add_simple_target!(:networks, switch.uid_ems)
```

Comment:
This is why the specs are currently failing, this should be `add_simple_target!(:switches, switch.uid_ems)`